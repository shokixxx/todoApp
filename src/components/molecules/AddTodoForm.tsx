import { Button } from '@mui/material'

import { InputField } from '../atmos/InputField'

type Props = {
  inputRef: React.RefObject<HTMLInputElement | null>
}

export const AddTodoForm = ({ inputRef }: Props) => {
  return (
    <div className="m-5 flex gap-5">
      <InputField ref={inputRef} className="w-2/3 p-5" />
      <Button variant="contained" type="submit" className="w-20">
        登録
      </Button>
    </div>
  )
}
