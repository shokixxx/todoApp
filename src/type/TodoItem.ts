export type TodoItem = {
  id: number
  name: string
  status: 'notStarted' | 'doing' | 'done'
}
