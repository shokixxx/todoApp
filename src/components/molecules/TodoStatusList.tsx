import DeleteIcon from '@mui/icons-material/Delete'
import IconButton from '@mui/material/IconButton'

import type { TodoItem } from '../../type/TodoItem'

type TodoStatusListProps = {
  statusTitle: string
  todoItems: TodoItem[]
}
export const TodoStatusList = ({ statusTitle, todoItems }: TodoStatusListProps) => {
  return (
    <>
      <h2 className="m-5 text-center text-2xl font-bold">{statusTitle}</h2>
      <ul>
        {todoItems.map((todoItem) => (
          <li
            key={todoItem.id}
            className="m-5 flex cursor-grab items-center justify-between rounded border bg-gray-600 p-4 hover:bg-gray-700"
            draggable
            onDragStart={(event) => event.dataTransfer.setData('text/plain', `${todoItem.id}`)}
          >
            {todoItem.name}
            <IconButton aria-label="delete">
              <DeleteIcon />
            </IconButton>
          </li>
        ))}
      </ul>
    </>
  )
}
