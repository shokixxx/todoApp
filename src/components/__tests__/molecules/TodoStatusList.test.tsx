import { fireEvent, render, screen, within } from '@testing-library/react'

import { TodoStatusList } from '../../molecules/TodoStatusList'

import type { TodoItem } from '../../../type/TodoItem'

const todoItems: TodoItem[] = [
  { id: 1, name: '本を買う', status: 'notStarted' },
  { id: 2, name: '課題を終わらせる', status: 'doing' },
  { id: 3, name: '健康診断を受ける', status: 'done' },
]

const mockFn = jest.fn()

// 参考: https://testing-library.com/docs/queries/byrole/#level
test('h2タグが渡されたステータスタイトルで表示されている', () => {
  render(<TodoStatusList statusTitle="未着手" todoItems={todoItems} handleDelete={mockFn} />)
  expect(screen.getByRole('heading', { level: 2, name: '未着手' })).toBeInTheDocument()
})

test('todoItems の数だけ一覧表示される', () => {
  render(<TodoStatusList statusTitle="未着手" todoItems={todoItems} handleDelete={mockFn} />)
  const list = screen.getByRole('list')
  expect(list).toBeInTheDocument()
  expect(within(list).getAllByRole('listitem')).toHaveLength(3)
})

test('Snapshot: todoItemsの数だけ一覧表示される', () => {
  const { container } = render(<TodoStatusList statusTitle="未着手" todoItems={todoItems} handleDelete={mockFn} />)
  expect(container).toMatchSnapshot()
})

test('削除ボタンがtodoItemsの数だけ表示される', () => {
  render(<TodoStatusList statusTitle="未着手" todoItems={todoItems} handleDelete={mockFn} />)
  expect(screen.getAllByLabelText('delete').length).toBe(3)
})

test('一番目の削除ボタンを押下すると、イベントハンドラーが呼ばれる', () => {
  render(<TodoStatusList statusTitle="未着手" todoItems={todoItems} handleDelete={mockFn} />)
  const deleteButtons = screen.getAllByLabelText('delete')
  fireEvent.click(deleteButtons[0])
  expect(mockFn).toHaveBeenCalled()
})
