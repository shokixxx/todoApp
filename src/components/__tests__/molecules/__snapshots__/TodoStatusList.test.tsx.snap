// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Snapshot: todoItemsの数だけ一覧表示される 1`] = `
<div>
  <h2
    class="m-5 text-center text-2xl font-bold"
  >
    未着手
  </h2>
  <ul>
    <li
      class="m-5 flex cursor-grab items-center justify-between rounded border bg-gray-600 p-4 hover:bg-gray-700"
      draggable="true"
    >
      本を買う
      <button
        aria-label="delete"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="DeleteIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
          />
        </svg>
      </button>
    </li>
    <li
      class="m-5 flex cursor-grab items-center justify-between rounded border bg-gray-600 p-4 hover:bg-gray-700"
      draggable="true"
    >
      課題を終わらせる
      <button
        aria-label="delete"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="DeleteIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
          />
        </svg>
      </button>
    </li>
    <li
      class="m-5 flex cursor-grab items-center justify-between rounded border bg-gray-600 p-4 hover:bg-gray-700"
      draggable="true"
    >
      健康診断を受ける
      <button
        aria-label="delete"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="DeleteIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
          />
        </svg>
      </button>
    </li>
  </ul>
</div>
`;
